### PR TITLE
New structural changes to test repo

### PR DESCRIPTION
**Pull request by nitishSr**
This PR tries to introduce new structure for organising test repo as per below structure - 
```
.
└── oadp-operator/
    └── tests/
        └── e2e/
            ├── templates/
            │   ├── velero_bsl_template.tmpl
            │   └── credentials
            ├── lib/
            │   ├── kube_helpers.go
            │   └── velero_helpers.go
            ├── utils/
            │   └── common.go
              velero_deployment_suite_test.go
              bsl_suite_test.go
              vsl_suite_test.go
              e2e_suite_test.go
```

### CODE REVIEW

**Comment by dymurray** [comment]
@nitishSr I like the structural changes, LGTM. Let me know when this is ready for review

File: tests/e2e/backup_restore_suite_test.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package e2e
+package e2e_test
```
Comment (shubham-pampattiwar) [inline tests/e2e/backup_restore_suite_test.go:1]: qq: Is the name change intentional or accidental ? The PR description does not mention about `e2e_test` package https://github.com/openshift/oadp-operator/pull/531#issue-1100332242

**Review by shubham-pampattiwar — APPROVED** [review]
@nitishSr +1 on the refactoring, just added one question, else LGTM.

**Comment by kaovilai** [comment]
+1
I think exporting functions would allow folks to reuse our e2e functions elsewhere.

File: tests/e2e/utils/common_utils.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package utils
+
+import (
+	"io/ioutil"
+)
+
+func ReadFile(path string) ([]byte, error) {
```
Comment (kaovilai) [inline tests/e2e/utils/common_utils.go:7]: I think we can remove this? This function doesn't do anything. Can remove later after merge. Thoughts?

**Review by kaovilai — APPROVED** [review]

File: tests/e2e/utils/common_utils.go (hunk 2 above)
Comment (kaovilai) [reply tests/e2e/utils/common_utils.go:7]: I see elsewhere we are using os.ReadFile as well which seems to do the same thing.

Comment (kaovilai) [reply tests/e2e/utils/common_utils.go:7]: Looks like os.ReadFile is more direct over io/ioutil now... at least from reading https://pkg.go.dev/io/ioutil@go1.17.6#ReadFile

File: tests/e2e/backup_restore_suite_test.go (hunk 1 above)
Comment (nitishSr) [reply tests/e2e/backup_restore_suite_test.go:1]: @shubham-pampattiwar yes that is intentional change. Just wanted to be more specific here, to reflect that its a test package (taken as reference from Ginkgo docs)

File: tests/e2e/utils/common_utils.go (hunk 2 above)
Comment (nitishSr) [reply tests/e2e/utils/common_utils.go:7]: @kaovilai I do think the same. I just kept it as it is, considering these changes were made by some earlier PR. I can definitely try to replace it altogether with os.ReadFile() if possible.

**Pull request merged**